frontend/src/contexts/searchContext.tsx

```diff
@@ -148,907 +148,908 @@ SearchContext.displayName = 'SearchContext';
+			if (
+				searchConfig.discussion_section !== null &&
+				/^[A-Z]*$/.test(listing.section)
+			) {
+				return false;
+			}
+
+			if (
+				searchConfig.fy_sem !== null &&
+				searchConfig.fy_sem !== listing.fysem
+			) {
+				return false;
+			}
+
+			if (
+				searchConfig.grad_level !== null &&
+				(listing.number === null ||
+					// tests if first character is between 5-9
+					(listing.number.charAt(0) >= "5" &&
+						listing.number.charAt(0) <= "9") ||
+					// otherwise if first character is not a number (i.e. summer classes), tests whether second character between 5-9
+					((listing.number.charAt(0) < "0" || listing.number.charAt(0) > "9") &&
+						(listing.number.length <= 1 ||
+							(listing.number.charAt(1) >= "5" &&
+								listing.number.charAt(1) <= "9"))))
+			) {
+				return false;
+			}
+
+			if (
+				searchConfig.subjects.size !== 0 &&
+				!searchConfig.subjects.has(listing.subject)
+			) {
+				return false;
+			}
+
+			const days = new Set(getDayTimes(listing)?.map((daytime) => daytime.day));
+			if (searchConfig.days.size !== 0) {
+				let include = true;
+				if (days && days !== null) {
+					days.forEach((day) => {
+						if (!searchConfig.days.has(day)) {
+							include = false;
+						}
+					});
+					searchConfig.days.forEach((day) => {
+						if (!days.has(day)) {
+							include = false;
+						}
+					});
+				} else {
+					include = false;
+				}
+				if (!include) {
+					return false;
+				}
+			}
+
+			if (
+				(searchConfig.areas.size !== 0 || searchConfig.skills.size !== 0) &&
+				!listing.areas.some((v: any): v is AreasType =>
+					searchConfig.areas.has(v)
+				) &&
+				!listing.skills.some((v: any): v is SkillsType =>
+					searchConfig.skills.has(v)
+				)
+			) {
+				return false;
+			}
+
+			if (
+				searchConfig.credits.size !== 0 &&
+				listing.credits !== null &&
+				!searchConfig.credits.has(String(listing.credits))
+			) {
+				return false;
+			}
+
+			if (
+				searchConfig.schools.size !== 0 &&
+				listing.school !== null &&
+				!searchConfig.schools.has(listing.school)
+			) {
+				return false;
+			}
+
+			// Handle search text. Each token must match something.
+			for (const token of tokens) {
+				// first character of the course number
+				const numberFirstChar = listing.number.charAt(0);
+				if (
+					listing.subject.toLowerCase().startsWith(token) ||
+					listing.number.toLowerCase().startsWith(token) ||
+					// for course numbers that start with a letter
+					// (checked by if .toLowerCase() is not equal to .toUpperCase(), see https://stackoverflow.com/a/32567789/5540324),
+					// exclude this letter when comparing with the search token
+					(numberFirstChar.toLowerCase() !== numberFirstChar.toUpperCase() &&
+						listing.number
+							.toLowerCase()
+							.startsWith(numberFirstChar.toLowerCase() + token)) ||
+					listing.title.toLowerCase().includes(token) ||
+					listing.description?.toLowerCase()?.includes(token) ||
```

Comment:
this is the line we want, right? Would it be possible to just commit this without the rest of the formatting changes? and maybe add it as a button with toggle on (if not I/someone else can work on doing that, shouldn't be hard).